xls/dslx/stdlib/apfloat.x

```diff
@@ -1481,14 +1481,15 @@ fn to_signed_or_unsigned_int<RESULT_SZ: u32, RESULT_SIGNED: bool, EXP_SZ: u32, F
         }
     };
 
+    type ReturnT = xN[RESULT_SIGNED][RESULT_SZ];
```

Comment:
I think the result is already the correct signedness.

Reply:
I thought this was the change you were trying to make in #1572? Maybe it was just for demonstration purposes -- I can revert this.

Reply:
Sorry for the confusion. I was trying to remove the cast on line 1507.

Reply:
The RESULT_SIGNED branch now returns sN and the other branch returns uN, which is made possible by the xN change. Thanks btw!

Reply:
No worries! Reverted the apfloat change here. In retrospect, separating that out probably makes sense regardless. I think this'll probably fix the root cause though. 👍

Reply:
Thanks, I'll fix up apfloat as a follow up.

Reply:
@cdleary where we probably got crossed is that I forgot DSLX doesn't have constexpr if's, so the types in both branches still need to match, while you probably knew that :)

My initial comment was thinking that we could write:

```
    if RESULT_SIGNED {
        // Reduce to the target size, preserving signedness.
        let result = result as sN[RESULT_SZ];
        if !x.sign { result } else { -result }
    } else {
        // Already unsigned, just size correctly.
        result as uN[RESULT_SZ]
    }
```

Anyway, submitted basically what you had with the additional cast in `to_int` removed in https://github.com/google/xls/commit/17d0d09be1f0c895adfd83478e91e728def009b9.

Reply:
You're giving me too much credit, was kind of blindly trying to do what you had described without thinking much. :-) Glad you were able to make the change based off this one though!